server/src/main/java/org/elasticsearch/index/mapper/NumberFieldMapper.java

```diff
@@ -266,9 +266,13 @@ public Float parse(XContentParser parser, boolean coerce) throws IOException {
             }
 
             @Override
-            public Query termQuery(String field, Object value) {
+            public Query termQuery(String field, Object value, boolean forceDocValues) {
```

Comment:
nit: I am not entirely happy with the naming of the new parameter. I think `forceDocValues` fits better for range queries where we use IndexOrDocValuesQuery, but here it's more about either one or the other. I also struggle slightly with always providing `isIndexed == false` to it. Shall we rather call it isIndexed for `termQuery` ? I find that that would express better the fact that the index has the precedence if present.

Reply:
ok, fixed in 4b273fd94c1